### ISSUE TITLE
Cascading Delete for Courses and Assignments

### ISSUE THREAD

**Issue by TanishG123** [kiat/codeAssist#89]
**Basic functional implementation done**

There were some issues with the regrade request and enrolment dependencies that needed to be taken care of, this is done.

**Future Implementation:**
Should consider adding cascading delete to various DB models to decrease work in the backend and ensure that we don't have to manually take care of various dependencies.

**Possible Dependencies for Cascading Delete:**
- Delete all **Enrollments** for a deleting **Course**
- Delete all **Assignments** for a deleting **Course**
- Delete all **Submissions** for a deleting **Assignment**
- Delete all **Regrade Requests** for a deleting **Submission**

**Comment by defkorean** [comment]
Was this not done in pr #82? Specifically the second and third bullet points?

**Comment by TanishG123** [comment]
Yes, all of the noted dependencies have been taken care of in either PR 82 or the PR for the resubmit process I put in. This is an alternative and more efficient approach to handle the same process and reduce the amount of nested loops and code needed in the backend - by instead adding a couple of statements into the actual db models.

**Comment by DarioDeSilva** [comment]
This is resolved. Course, Assignment, and Submission models now use ondelete="CASCADE" for Enrollments/Assignments, Submissions, and RegradeRequests/TestCaseResults respectively (backend/api/models.py). Closing as implemented at the DB level, per the approach discussed in the comments.
